api/test/integration/test_agentsPUT_endpoints.tavern.yaml

```diff
@@ -296,6 +296,88 @@ stages:
       json:
         code: 1114
 
+---
+test_name: PUT /groups/{group_id}/restart
+
+stages:
+
+    # PUT /groups/group1/restart (This group has no agents assigned)
+  - name: Restart all agents from group1
+    request:
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/{version:s}/groups/group1/restart"
+      method: PUT
+      headers:
+        Authorization: "Bearer {test_login_token}"
+        content-type: "application/xml"
+    response:
+      status_code: 400
+      json:
+        title: "Wazuh Error"
+        detail: "The group does not have any agent assigned"
+        status: 400
+        remediation: "Please select another group or assign any agent to it"
+        code: 1755
+        dapi_errors:
+          master-node:
+            error: "The group does not have any agent assigned"
+
+    # PUT /groups/group2/restart (Agents 009 and 010 are disconnected)
+  - name: Restart all agents from group2
+    request:
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/{version:s}/groups/group2/restart"
+      method: PUT
+      headers:
+        Authorization: "Bearer {test_login_token}"
+        content-type: "application/xml"
+    response:
+      status_code: 200
+      json:
+        data:
+          affected_items:
+            - "001"
+            - "002"
+            - "003"
+            - "004"
+            - "005"
+            - "006"
+            - "007"
+            - "008"
+          total_affected_items: 8
+          failed_items:
+            - error:
+                code: 1707
+                message: "Impossible to restart non-active agent: disconnected"
+                remediation: "Please, make sure agent is active before attempting to restart"
+              id:
+                - "009"
+                - "010"
+          total_failed_items: 2
+        message: "Could not send command to some agents"
+    delay_after: 45
+
+  # PUT /groups/not_exists/restart (Group does not exist)
+  - name: Restart all agents from a group which does not exist
+    request:
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/{version:s}/groups/not_exists/restart"
+      method: PUT
+      headers:
+        Authorization: "Bearer {test_login_token}"
+        content-type: "application/xml"
+    response:
+      status_code: 400
+      json:
```

Comment:
I would remove all checks except code 1710 to avoid excess compromise with the messages.